R/filter_occs_time.R

```diff
@@ -0,0 +1,277 @@
+#' absolute stratigraphic ages for filtering fossil occurrences, such as that
+#' returned by [time_bins()]. This `data.frame` must contain at least
+#' three columns as named by `interval_name` (defaults to "interval_name"),
+#' `max_ma` (defaults to "max_ma") and `min_ma` (defaults to "min_ma"). Columns
+#' `max_ma` and `min_ma` must be numeric.
+#' @param interval_name `character` The name of the column in `bins` that
+#' contains the interval names.
+#' @param youngest_bin `character`. The name of the youngest stratigraphic
+#' interval to trim to. Only one interval name should be supplied, which should
+#' be present in column `interval_name` in `bins`.
+#' @param oldest_bin `character`. The name of the oldest stratigraphic interval
+#' to trim to. Only one interval name should be supplied, which should be
+#' present in column `interval_name` in `bins`.
+#' @param min_age `numeric`. The minimum absolute occurrence age to trim to.
+#' Ignored if `youngest_bin` is supplied.
+#' @param max_age `numeric`. The maximum absolute occurrence age to trim to.
+#' Ignored if `oldest_bin` is supplied.
+#' @param uncertainty A numeric value denoting the maximum tolerable temporal
+#' uncertainty for retaining occurrences. Alternatively, if this argument is set
+#' to "bin", then only occurrences which span no more than `nbin` intervals in
+#' `bins` will be retained.
+#' @param nbin `numeric` The maximum tolerable bin-wise uncertainty above which occurrences
+#' will be discarded if `uncertainty = "bin"` (defaults to `1`).
+#' @param retain.taxa `logical`. Complete loss of some taxa may occur if all
+#' their occurrences exceed the temporal uncertainty criterion set by the user.
+#' If this is undesirable, occurrences for these taxa will be retained as
+#' exceptions to the filtering parameters (defaults to `FALSE`).
+#' @param taxa `character`. The name of the column in `occdf` you wish to be
+#' treated as the taxon names in `occdf` if `retain.taxa = TRUE` (defaults to
+#' `NULL`).
+#'
+#' @returns
+#' The `data.frame` containing the original input `occdf`, minus any
+#' occurrences discarded under the chosen trimming and filtering parameters. If
+#' `retain.taxa = TRUE`, there will be an additional column called `retained`,
+#' consisting of `0` for occurrences which met the filtering criteria and `1`
+#' for occurrences which were retained for taxonomic completeness. Depending on
+#' the filtering parameters used, it is possible that all or no occurrences
+#' will be discarded, resulting in an empty `data.frame` or an unchanged data
+#' set respectively.
+#'
+#' @details
+#' When trimming by stratigraphic interval names, the boundary ages of
+#' `youngest_bin` and `oldest_bin` are taken from their entries in `bins`.
+#' Occurrences with maximum ages younger than or equal to the upper boundary
+#' age of `youngest_bin` or minimum ages older than or equal to the lower
+#' boundary age of `oldest_bin` will be discarded.
```

Comment:
Unless I'm mistaken, I think this better matches what the code is doing here?
https://github.com/jf15558/palaeoverse/blob/b8df3dd76700c9f0532ae5c84c4c45dfe22145b3/R/filter_occs_time.R#L215-L216

```suggestion
#' Occurrences with maximum ages younger than the upper boundary
#' age of `youngest_bin` or minimum ages older than the lower
#' boundary age of `oldest_bin` will be discarded.
```